### PR TITLE
Update docstrings and add doctests for task-related functions

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
@jameson or @amitmurthy (not sure who to ping), this look ok?

File: base/event.jl [hunk 1]
```diff
@@ -85,11 +85,28 @@ schedule(t::Task) = enq_work(t)
     schedule(t::Task, [val]; error=false)
 
 Add a task to the scheduler's queue. This causes the task to run constantly when the system
-is otherwise idle, unless the task performs a blocking operation such as `wait`.
+is otherwise idle, unless the task performs a blocking operation such as [`wait`](:func:`wait`).
 
 If a second argument `val` is provided, it will be passed to the task (via the return value of
-`yieldto`) when it runs again. If `error` is `true`, the value is raised as an exception in
+[`yieldto`](:func:`yieldto`) when it runs again. If `error` is `true`, the value is raised as an exception in
```
Comment (tkelman) [inline base/event.jl]: need another close paren

**Comment by kshyatt** [comment]
Error in `pkg` tests looks unrelated. Ok to squash and merge?

File: base/task.jl [hunk 2]
```diff
@@ -55,6 +55,20 @@ end
...
+julia> a1() = det(rand(1000, 1000));
+
+julia> b = @task a1();
+
+julia> istaskstarted(b)
+false
+
+julia> schedule(b);
+
+julia> istaskdone(b)
```
Comment (vtjnash) [inline base/task.jl]: When run as a doctest, this should return false. Adding an explicit 'yield()' statement should fix that (due to fair scheduling)

Comment (vtjnash) [inline base/task.jl]: Or maybe you wanted istaskstarted?

Comment (kshyatt) [inline base/task.jl]: I want to see that the task is done after it's run by `schedule`

Comment (amitmurthy) [inline base/task.jl]: A `yield()` should do it. `schedule` adds it to the end of runnable task queue. Yielding will enable it to be run and hence complete.

**Comment by kshyatt** [comment]
@vtjnash good to go if CI passes?

File: doc/stdlib/parallel.rst [hunk 3]
```diff
@@ -93,6 +126,50 @@ Tasks
...
+
+       julia> schedule(b);
+
+       julia> istaskstarted(b)
+       true
+
+       julia> istaskdone(b)
+       true
+
+<<<<<<< HEAD
```
Comment (vtjnash) [inline doc/stdlib/parallel.rst]: rebase error?

File: base/event.jl [hunk 4]
```diff
@@ -90,6 +90,23 @@ is otherwise idle, unless the task performs a blocking operation such as [`wait`
...
+```jldoctest
+julia> a5() = det(rand(1000, 1000));
+
+julia> b = Task(a5);
+
+julia> istaskstarted(b)
+false
+
+julia> schedule(b);
+
```
Comment (vtjnash) [inline base/event.jl]: should probably stick `yield()` here too

**Comment by kshyatt** [comment]
I think I got rid of the rebase errors and the OSX thing seems not my fault?

**Comment by kshyatt** [comment]
Bump.

**Pull request merged**